scripts/testing/estimate_calories_from_image.py

```diff
@@ -0,0 +1,318 @@
+                     food_side_bounding_box, 
+                     food_top_bounding_box, 
+                     coin_side_bounding_box, 
+                     coin_top_bounding_box,
+                     food_label):
+
+  # compute scaling factor for coin side and top
+  side_image_width_px, side_image_height_px, side_scaling_factor, food_side_width_px, food_side_height_px = \
+    get_measurements(side_image_np, coin_side_bounding_box, food_side_bounding_box)
+  top_image_width_px, top_image_height_px, top_scaling_factor, food_top_width_px, food_top_height_px = \
+    get_measurements(top_image_np, coin_top_bounding_box, food_top_bounding_box)                                                                             
+  
+  # compute bounds for foods
+  food_side_bounds = (int(food_side_bounding_box[0] * side_image_width_px), \
+                      int(food_side_bounding_box[1] * side_image_height_px), \
+                      int(food_side_width_px), \
+                      int(food_side_height_px))
+
+  food_top_bounds = (int(food_top_bounding_box[0] * top_image_width_px), \
+                     int(food_top_bounding_box[1] * top_image_height_px), \
+                     int(food_top_width_px), \
+                     int(food_top_height_px))
+
+  # compute volume
+  volume = compute_volume_with_grabcut(side_image_np, 
+                                       top_image_np, 
+                                       food_side_bounds, 
+                                       food_top_bounds, 
+                                       side_scaling_factor, 
+                                       top_scaling_factor,
+                                       food_label)
+
+  # compute calories
+  return volume * FOOD_CALORIE[food_label]
+
+
+
+def test():
+  # helper functions for test
+  def load_image_into_numpy_array(path, w, h):
+    import tensorflow as tf
+    from six import BytesIO
+    """Load an image from file into a numpy array.
+
+    Puts image into numpy array to feed into tensorflow graph.
+    Note that by convention we put it into a numpy array with shape
+    (height, width, channels), where channels=3 for RGB.
+
+    Args:
+      path: the file path to the image
+
+    Returns:
+      uint8 numpy array with shape (img_height, img_width, 3)
+    """
+    image = None
+    if(path.startswith('http')):
+      response = urlopen(path)
+      image_data = response.read()
+      image_data = BytesIO(image_data)
+      image = Image.open(image_data)
+    else:
+      image_data = tf.io.gfile.GFile(path, 'rb').read()
+      image = Image.open(BytesIO(image_data))
+
+    bsize = [w, h]
+    image.thumbnail(bsize, Image.ANTIALIAS)
+    (im_width, im_height) = image.size
+    print("Resized Width {}  Height {}".format(im_width, im_height))
+    return np.array(image.getdata()).reshape(
+        (1, im_height, im_width, 3)).astype(np.uint8)
+
+  from matplotlib.image import imread
+  
+  # create numpy image
+  w, h = 800, 600
```

Comment:
Wait, are you using this instead of resizing the images manually? Or I'm assuming that's what you're doing to make the images like 100 pixels instead of like 600, for grabcutting?

Reply:
my bad, moved this testing code out to a different file